src/drivers/amd/vangogh/acp_dmic_dma.c

```diff
@@ -0,0 +1,218 @@
+		 0xb9, 0x42, 0x59, 0xe2, 0x0a, 0x66, 0x11, 0xbe);
+DECLARE_TR_CTX(acp_dmic_dma_vgh_tr, SOF_UUID(acp_dmic_dma_uuid), LOG_LEVEL_INFO);
+
+uint32_t dmic_rngbuff_size;
+
+int acp_dmic_dma_start(struct dma_chan_data *channel)
+{
+	acp_wov_pdm_no_of_channels_t pdm_channels;
+	acp_wov_pdm_decimation_factor_t deci_fctr;
+	acp_wov_misc_ctrl_t wov_misc_ctrl;
+	acp_wov_pdm_dma_enable_t  pdm_dma_enable;
+	acp_hstdm_iter_t        hs_iter;
+	acp_hstdm_irer_t        hs_irer;
+	uint32_t                acp_pdm_en;
+
+	struct timer *timer = timer_get();
+	uint64_t deadline = platform_timer_get(timer) +
+				clock_ms_to_ticks(PLATFORM_DEFAULT_CLOCK, 1) * 500 / 1000;
+	 hs_iter = (acp_hstdm_iter_t)io_reg_read((PU_REGISTER_BASE + ACP_HSTDM_ITER));
+	hs_irer = (acp_hstdm_irer_t)io_reg_read((PU_REGISTER_BASE + ACP_HSTDM_IRER));
+	acp_pdm_en = (uint32_t)io_reg_read(PU_REGISTER_BASE + ACP_WOV_PDM_ENABLE);
+
+	if (!hs_iter.bits.hstdm_txen && !hs_irer.bits.hstdm_rx_en && !acp_pdm_en) {
+		io_reg_write((PU_REGISTER_BASE + ACP_CLKMUX_SEL), ACP_ACLK_CLK_SEL);
+		/* Request SMU to set aclk to 600 Mhz */
+		acp_change_clock_notify(600000000);
+	}
+	channel->status = COMP_STATE_ACTIVE;
+	if (channel->direction == DMA_DIR_DEV_TO_MEM) {
+		/* Channel for DMIC */
+		pdm_channels.bits.pdm_no_of_channels = 0;
+		io_reg_write(PU_REGISTER_BASE + ACP_WOV_PDM_NO_OF_CHANNELS,
+			     pdm_channels.u32all);
+		/* Decimation Factor */
+		deci_fctr.u32all = 2;
+		io_reg_write(PU_REGISTER_BASE + ACP_WOV_PDM_DECIMATION_FACTOR,
+			     deci_fctr.u32all);
+		/* PDM Control */
+		wov_misc_ctrl = (acp_wov_misc_ctrl_t)
+				io_reg_read(PU_REGISTER_BASE + ACP_WOV_MISC_CTRL);
+		wov_misc_ctrl.u32all |= 0x10;
+		io_reg_write(PU_REGISTER_BASE + ACP_WOV_MISC_CTRL,
+			     wov_misc_ctrl.u32all);
+		/* PDM Enable */
+		io_reg_write(PU_REGISTER_BASE + ACP_WOV_PDM_ENABLE, 1);
+		/* PDM DMA Enable */
+		io_reg_write(PU_REGISTER_BASE + ACP_WOV_PDM_DMA_ENABLE, 1);
+		/* Check the PDM DMA Enable status bit */
+		pdm_dma_enable = (acp_wov_pdm_dma_enable_t)
+				 io_reg_read(PU_REGISTER_BASE + ACP_WOV_PDM_DMA_ENABLE);
+		while (!(uint32_t)pdm_dma_enable.bits.pdm_dma_en_status) {
+			if (deadline < platform_timer_get(timer)) {
+				/* safe check in case we've got preempted after read */
+				if ((uint32_t)pdm_dma_enable.bits.pdm_dma_en_status)
+					return 0;
+				tr_err(&acp_dmic_dma_vgh_tr, "timed out for dma start");
+				return -ETIME;
+			}
+			pdm_dma_enable = (acp_wov_pdm_dma_enable_t)
+					io_reg_read(PU_REGISTER_BASE + ACP_WOV_PDM_DMA_ENABLE);
+		}
+	}
+	return 0;
+}
+
+int acp_dmic_dma_stop(struct dma_chan_data *channel)
+{
+	acp_wov_pdm_dma_enable_t  pdm_dma_enable;
+	acp_hstdm_iter_t        hs_iter;
+	acp_hstdm_irer_t        hs_irer;
+	uint32_t acp_pdm_en;
+	struct timer *timer = timer_get();
+	uint64_t deadline = platform_timer_get(timer) +
+			clock_ms_to_ticks(PLATFORM_DEFAULT_CLOCK, 1) * 500 / 1000;
+	switch (channel->status) {
```

Comment:
checkpatch usually complains when there's no empty line between variable definitions and code